### ISSUE TITLE
get uploadparamsfortemplate results in NPE 

### ISSUE THREAD

**Issue by borisstoyanov** [apache/cloudstack#6071]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.15.2
~~~

##### CONFIGURATION

VMWare 

##### SUMMARY

When executing the following API call we get an NPE, it was observed the same to be working in previous versions

##### STEPS TO REPRODUCE

~~~
(localcloud) 🐱 > get uploadparamsfortemplate displaytext=test format=ova hypervisor=vmware name=test ostypeid=c062e1c5-72ec-11ec-bf30-1e00310003f0 zoneid=8e5812f7-282a-46a1-a7ac-b597cc060277
~~~

##### EXPECTED RESULTS

~~~
Get a valid response 
~~~

##### ACTUAL RESULTS

~~~
2022-03-08 14:15:03,354 DEBUG [c.c.a.ApiServer] (qtp1762902523-20:ctx-9f316b84 ctx-c4a676b3) (logid:c2199583) CIDRs from which account 'Acct[dcb3516f-72ec-11ec-bf30-1e00310003f0-admin]' is allowed to perform API calls: 0.0.0.0/0,::/0
2022-03-08 14:15:03,357 ERROR [c.c.a.ApiServer] (qtp1762902523-20:ctx-9f316b84 ctx-c4a676b3 ctx-99ad693b) (logid:c2199583) unhandled exception executing api command: [Ljava.lang.String;@66d405bd
java.lang.NullPointerException
	at org.apache.cloudstack.api.command.user.template.GetUploadParamsForTemplateCmd.validateRequest(GetUploadParamsForTemplateCmd.java:186)
	at org.apache.cloudstack.api.command.user.template.GetUploadParamsForTemplateCmd.execute(GetUploadParamsForTemplateCmd.java:168)
	at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:156)
	at com.cloud.api.ApiServer.queueCommand(ApiServer.java:764)
	at com.cloud.api.ApiServer.handleRequest(ApiServer.java:588)
	at com.cloud.api.ApiServlet.processRequestInContext(ApiServlet.java:321)
	at com.cloud.api.ApiServlet$1.run(ApiServlet.java:134)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:102)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52)
	at com.cloud.api.ApiServlet.processRequest(ApiServlet.java:131)
	at com.cloud.api.ApiServlet.doGet(ApiServlet.java:93)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1443)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:766)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-03-08 14:15:03,358 DEBUG [c.c.a.ApiServlet] (qtp1762902523-20:ctx-9f316b84 ctx-c4a676b3 ctx-99ad693b) (logid:c2199583) ===END===  10.0.35.93 -- GET  apiKey=LIN6rqXuaJwMPfGYFh13qDwYz5VNNz1J2J6qIOWcd3oLQOq0WtD4CwRundBL6rzXToa3lQOC_vKjI3nkHtiD8Q&command=getUploadParamsForTemplate&displaytext=test&format=ova&hypervisor=vmware&name=test&ostypeid=c062e1c5-72ec-11ec-bf30-1e00310003f0&response=json&zoneid=8e5812f7-282a-46a1-a7ac-b597cc060277&signature=5m4f79Q5VjvyDPI%2ByETamRdsaIY%3D
~~~

**Comment by nvazquez** [comment]
@Pearl1594 I think we have fixed this issue before, do you recall?

**Comment by Pearl1594** [comment]
Is it this you are referring to - https://github.com/apache/cloudstack/pull/5468? If so, that PR doesn't solve the issue. And I don't recollect any other fix for this. As Daan mentioned, it seems as if hypervisor field wasn't passed - however, it's a required parameter.

**Comment by nvazquez** [comment]
I see its been passed `hypervisor=vmware`, I think the issue comes from the deployAsIs value. @borisstoyanov can you try passing `deployasis=false`

**Comment by nvazquez** [comment]
Have fixed Vmware template upload on PR https://github.com/apache/cloudstack/pull/4555

**Comment by nvazquez** [comment]
Fixed on #6079
